flight/PiOS/inc/pios_hal.h

```diff
@@ -3,20 +3,27 @@
 
 #include <uavobjectmanager.h> /* XXX TODO */
 #include <hwshared.h>
+
+#ifdef PIOS_INCLUDE_USART
 #include <pios_usart_priv.h>
-#include <pios_sbus_priv.h>
-#include <pios_dsm_priv.h>
-#include <pios_ppm_priv.h>
-#include <pios_pwm_priv.h>
-#include <pios_i2c_priv.h>
-#include <pios_usb_cdc_priv.h>
-#include <pios_usb_hid_priv.h>
+#endif
 
 #if defined(PIOS_INCLUDE_RFM22B)
 #include <pios_rfm22b_priv.h>
 #include <pios_openlrs_priv.h>
 #endif /* PIOS_INCLUDE_RFM22B */
 
+struct pios_dsm_cfg;
```

Comment:
I punted on almost all of this.  Only a few were doable, and the ones that were seemed like a bad idea (other than pios_com_driver).

I will start a series of PRs to clean up individual headers in the style of pios_flyingpio.h.  And it's pretty clear that pios_hal is not general enough and needs some reimagining, since it's very hacked here.